mantidproject/mantid
# Split Comboboxes in EA auto tab

**Description of work.**
All comboboxes where users can select a workspace or table have been split into 2 comboboxes sorting them by their runs

**Report to:** Adrian Hillier


**To test:**
1. Before starting Mantid go to https://github.com/mantidproject/mantid/blob/master/Framework/Properties/Mantid.properties.template#L24 , add Muon/Elemental_Analysis_2.py to the end of the line and build Mantid.

2. Download the test files from PR #29958 and add to User Directories.

3. Open the elemental analysis 2 GUI and load 9997.

4. Open the automatic tab and next to the find peak button there should be 2 comboboxes on the left the first combobox  should show 9997 and the second combobox should show All ,detector 1, detector 2, detector 3, detector 4.
5. Load 9998 and in the first combox there should be 9997 and 9998 in the first combobox.
6. Removing a detector in the grouping table causes that option to not be available in second combobox when run is selected
7. When any tables are produced they are sorted by run number in comboboxes beside show matches and show peaks button

Fixes #31729 . 


*This does not require release notes* because Elemental analysis GUI is not yet visible to users in Mantid Workbench

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
